src/app.tsx

```diff
@@ -164,10 +164,22 @@ class AppState extends EventEmitter<AppStateEvents> {
         this.#initPromise = doit();
         return this.#initPromise;
     }
+
+    // Asynchronously wait until the list of VMs has been loaded, and
+    // then return it.
+
+    async getVms(): Promise<VM[]> {
+        await this.init();
```

Comment:
So this seems to not do what I would think it does?

Looking at the implementation of `init()` it returns a promise, but also tries to do caching so we don't call `doit()` all the time. But it seems that in the end that is exactly what happens?

Shouldn't you assign `this.#initPromise = Promise.resolve(void)` and await doit() in `init()`. Then you only call `doit()` once.. now you return it as promise to be awaited everywhere we call `getVms()`